test/Main.hs

```diff
@@ -651,6 +651,12 @@ streamOperations (stream, list, len) = do
             return $ str == lst
         `shouldReturn` True
 
+    it "iterate" $ do
+            let lst = take len $ iterate (+ 1) 0
+            str <- A.toList . serially . (A.take len) $ (A.iterate (+ 1) (0 :: Int))
+            return $ str == lst
+        `shouldReturn` True
```

Comment:
Since `lst` is not monadic we write that directly on the RHS of `shouldReturn`:

```
A.toList . serially . (A.take len) $ (A.iterate (+ 1) (0 :: Int)) `shouldReturn`  take len $ iterate (+ 1) 0
```